Draft of the ThanosCon blog post. (#42)
bwplotka/my: web/content/posts/thanoscon2024.md

ThanosCon was held as a co-located KubeConEU 2024 half-day on March 19th in Paris. This means to get into the event you had to buy or get (e.g. via sponsoring or speaking at KubeCon or co-located event) the "All-Access pass", which on top of the 3-day KubeCon, got you access to **any** of the 19 co-located events day before. The specificality of being a co-located event is worth noting. There were essentially ~6000 attendees (estimation, [the official report will come](https://www.cncf.io/reports/)) on 19th March that could choose any of those 19 co-located events (up to the room capacity). In theory, this means that in every talk from a single even, you could have a different audience.

ThanosCon was scheduled in the morning part of the day (9:00-13:00) in a convenient room (noise isolation, chairs-to-space ratio, AV setup, huge screen) with a capacity of 150 ppl.

![Colin Douch sharing learnings from running Thanos on an extreme scale in Cloudflare to the full room!](/images/blog/thanoscon/thanoscon3.jpg)

> [Colin Douch](https://twitter.com/sinkingpoint) sharing learnings from running Thanos on an extreme scale in Cloudflare to the full room!

> I love how the ThanosCon went. We got a "family" vibe atmosphere with people feeling safe to ask (also difficult) questions during the Q&A and afterwards. I also loved how our event was fully focused on one topic--scaling your metrics, cheaply and reliably with the Thanos project. It feels like it allowed the audience to spend the beginning of their day productively, without many distractions, discussing one aspect deeply, end to end. Everyone learned, especially the Thanos contributors and team how to move the project forward.

Reviewer: ```suggestion
I love how the ThanosCon went. We got a "family" vibe atmosphere with people feeling safe to ask (also difficult) questions during the Q&A and afterwards. I also loved how our event was fully focused on one topic--scaling your metrics, cheaply and reliably with the Thanos project. It feels like it allowed the audience to spend the beginning of their day productively, without many distractions, discussing one aspect deeply, end to end. Everyone learned new things, especially the Thanos contributors and team, and got ideas on how to move the project forward.
```

* Intro - Michael Hoffmann, Aiven; Bartłomiej Płotka Google
* 6 Learnings from Building Thanos Project - Bartłomiej Płotka, Google; Fabian Reinartz, Databricks
* **⚠️** Monitoring the World: Scaling Thanos in Dynamic Prometheus Environments - Colin Douch, Cloudflare
* **⚠️** Scaling Thanos at Reddit - Ben Kochie & Trevor Riles, Reddit
* Thanos Project Update - Saswata Mukherjee, Red Hat
* Connecting Thanos to the Outer Rim via Query API - Filip Petkovski, Shopify & Mahmoud Amin, Google
* **⚠️** Multiverse of Thanos: Making Thanos Multi-Tenanted - Jacob Baungård Hansen & Coleen Iona Quadros, Red Hat
* Thanos Receiver Deep Dive - Joel Verezhak, Open Systems
* Closing Remarks - Harshitha Chowdary T, Amazon; Bartłomiej Płotka Google

> The speaker diversity across companies was great, but across gender, it was not. We tried hard, but only 22% of talks had a non-male identifying person co-speaking (supposed to be 33%, but last-minute changes due to external reasons occurred). Only 14% if you exclude opening and closing talks.

Reviewer: Super important to note! I wonder if we need to consider other DEI factors as well?

Author: I wished, but I don't have that data really. Let's start small?

> ThanosCon was exclusively sponsored by the CNCF. Kudos to the CNCF, but not ideal--it's not how this works on KubeCon. The chances of this event happening again without external sponsors are low. We contacted a few Thanos big end-users, but the majority of feedback was "oh it's December, Xmas time, decision makers not responding" or "we already closed the sponsors budgets for this quarter, if only you let us know in advance". If you want to see this event again, help us with some leads/motivate your employer! (: On the bright side this is typical for first-time events and not unique compared to other co-located events on that day as well.

Reviewer: ```suggestion
ThanosCon was exclusively sponsored by the CNCF. Kudos to the CNCF, but not ideal--this is not how it works on KubeCon. The chances of this event happening again without external sponsors are low. We contacted a few Thanos big end-users, but the majority of feedback was "oh it's December, Xmas time, decision makers not responding" or "we already closed the sponsors budgets for this quarter, if only you let us know in advance". If you want to see this event again, help us with some leads/motivate your employer! (: On the bright side this is typical for first-time events and not unique compared to other co-located events on that day as well.
```

On PromCon in late September 2023 in Berlin, organized by amazing folks from [PolarSignals](https://www.polarsignals.com) (and the CNCF) I started to probe ThanosCon idea among the present members of the Thanos community. It would be useful for you to understand that historically, PromCon was always **a one-track, two-day ultra-focused conference** which everyone enjoyed--the topics were focused, the event small enough to be able to talk to people you know already and meet new ones. To be fair, as the project grows, even PromCon gets to talk about the wider ecosystem and broadening the spectrum of topics--while in the past it was all about using Prometheus binary for metric monitoring and observability, now we talk also about external dashboards, AIOps, OpenTelemetry integrations, and of course "distributed Prometheus" systems like Thanos, Cortex or Mimir, which treats Prometheus mostly as an "agent/collector".

> The same happens on the Observability Day, and Prometheus Day events and so on. In each of those events we had one or two Thanos talks, but if you want to focus your brain on deeper Thanos usage, best practices and setups, making connections with the community and sharing your experience, bad luck -- you need to sit down and listen (and switch contexts) about other, potentially unrelated to you systems or observability signals. Don't get me wrong, there is value in huge 10k+ ppl events, with a dozen tracks and a thousand topics, but the proposition is a bit different, you are not there to focus on one e.g. system deeply.

Reviewer: Maybe `...` would be nicer break than comma
```suggestion
The same happens on the Observability Day, and Prometheus Day events and so on. In each of those events we had one or two Thanos talks, but if you want to focus your brain on deeper Thanos usage, best practices and setups, making connections with the community, and sharing your experience ... bad luck -- you need to sit down and listen (and switch contexts) about other, potentially unrelated to you systems or observability signals. Don't get me wrong, there is value in huge 10k+ ppl events, with a dozen tracks and a thousand topics, but the proposition is a bit different, you are not there to focus on one e.g. system deeply.
```

Finally, I had an additional selfish reason for organizing the ThanosCon 🙃. I wanted to create an excuse to spend some time with [Fabian Reinartz](https://twitter.com/fabxc). The last time we truly spent time together was during the Prometheus London meetup in 2018 where we introduced the initial design and implementation of the Thanos project we created in 3 months. ThanosCon was an epic moment to stop and reflect on what we learned while observing Thanos's growth in the open source. If you are interested, you are welcome to watch our 15m talk ["6 Learnings from Building Thanos Project - Bartłomiej Płotka, Google; Fabian Reinartz, Databricks"](https://youtu.be/ur8dDFaNEFg?si=0jYwzumqkI6e4dpF&t=350). Those were honest lessons that I plan to be aware of in my future work too (:

### Preparations

At the end of the day, we got chosen (yay!) by the CNCF. Officially the co-chairs of the event were [Sonia Singla](https://twitter.com/soniasinglas) and me, but in practice we had massive help from other members of the Thanos community and CNCF (more on that in [here](#gratitude)).

> We started by gathering [10 people](#gratitude) who would help the Programme Committee around Nov 27th--helping us to select the talks and finalize the schedule. In the meantime, we [wrote a blog post](https://thanos.io/blog/2023-20-11-thanoscon/) started an activity on social media and contacted a few end users for talks and sponsorship.

Reviewer: ```suggestion
We started by gathering [10 people](#gratitude) who would help the Programme Committee around Nov 27th--helping us to select the talks and finalize the schedule. In the meantime, we [wrote a blog post](https://thanos.io/blog/2023-20-11-thanoscon/) started actively sharing about it on social media and contacted a few end users for talks and sponsorship.
```

> After the CFP period finished (Dec 3th), from Dec 5th to 13th, the PC spent to reviewing the talks. We got a lot of amazing proposals, and it was pretty difficult to reject the majority of them, but that's how it works. Hope to see most of them at another event!

Reviewer: ```suggestion
After the CFP period finished (Dec 3th), from Dec 5th to 13th, the PC spent time reviewing the talks. We got a lot of amazing proposals, and it was pretty difficult to reject the majority of them, but that's how it works. Hope to see most of them at another event!
```

I would like to start by thanking the wide ThanosCon organizing team for amazing work:
* [Sonia Singla](https://twitter.com/soniasinglas) sad you couldn't make it in person, but you helped remotely massively ❤️
* [Filip Petkovski](https://twitter.com/fpetkovsky) thanks for your intuition and especially amazing work on mentoring others 😉! I know now where this "Staff" is coming from (:
* [Harshitha Chowdary Thota](https://twitter.com/ThotaHarshitha) the last-minute travel and preparations to coordinate ThanosCon in-person, with travel paid only by yourself--super impressive, lots of courage and dedication! Thank you!
* [Matej Gera](https://twitter.com/gmat90) thanks for your dedication during preparations and the schedule finalization.
* [Michael Hoffman](https://github.com/MichaHoffmann) I really learned a lot from your speaking skills, humour and high-disciplined preparation skills!

> * [Saswata Mukherjee](https://twitter.com/saswatamcode) words are not enough to express the amount of dedication and work you did under the hood for ThanosCon, plus amazing project updates talk, thanks! As always, inspiring others with passion.

Reviewer: Thank you! ❤️
